frontend/src/FirstRun/AuthenticationRequiredModalContent.tsx

```diff
@@ -0,0 +1,188 @@
+                helpText={translate('AuthenticationMethodHelpText')}
+                helpTextWarning={
+                  authenticationMethod.value === 'none'
+                    ? translate('AuthenticationMethodHelpTextWarning')
+                    : undefined
+                }
+                helpLink="https://wiki.servarr.com/sonarr/faq#forced-authentication"
+                onChange={onInputChange}
+                {...authenticationMethod}
+              />
+            </FormGroup>
+
+            <FormGroup>
+              <FormLabel>{translate('AuthenticationRequired')}</FormLabel>
+
+              <FormInputGroup
+                type={inputTypes.SELECT}
+                name="authenticationRequired"
+                values={authenticationRequiredOptions}
+                helpText={translate('AuthenticationRequiredHelpText')}
+                onChange={onInputChange}
+                {...authenticationRequired}
+              />
+            </FormGroup>
+
+            <FormGroup>
+              <FormLabel>{translate('Username')}</FormLabel>
+
+              <FormInputGroup
+                type={inputTypes.TEXT}
+                name="username"
+                helpTextWarning={
+                  username?.value
+                    ? undefined
+                    : translate('AuthenticationRequiredUsernameHelpTextWarning')
+                }
+                onChange={onInputChange}
+                {...username}
+              />
+            </FormGroup>
+
+            <FormGroup>
+              <FormLabel>{translate('Password')}</FormLabel>
+
+              <FormInputGroup
+                type={inputTypes.PASSWORD}
+                name="password"
+                helpTextWarning={
+                  password?.value
+                    ? undefined
+                    : translate('AuthenticationRequiredPasswordHelpTextWarning')
+                }
+                onChange={onInputChange}
+                {...password}
+              />
+            </FormGroup>
+
+            <FormGroup>
+              <FormLabel>{translate('PasswordConfirmation')}</FormLabel>
+
+              <FormInputGroup
+                type={inputTypes.PASSWORD}
+                name="passwordConfirmation"
+                helpTextWarning={
+                  passwordConfirmation?.value
+                    ? undefined
+                    : translate(
+                        'AuthenticationRequiredPasswordConfirmationHelpTextWarning'
+                      )
+                }
+                onChange={onInputChange}
+                {...passwordConfirmation}
+              />
+            </FormGroup>
+          </div>
+        ) : null}
+
+        {!isPopulated && !error ? <LoadingIndicator /> : null}
+      </ModalBody>
+
+      <ModalFooter>
+        <SpinnerButton
+          kind={kinds.PRIMARY}
+          isSpinning={isSaving}
+          isDisabled={!authenticationEnabled}
+          onPress={() => {
```

Comment:
This will recreate the function each time the component renders, whereas `useCallback` will not.